docs/queue.md

```diff
@@ -335,6 +335,22 @@ Note you can use [partial batch failures](#partial-batch-failures) to avoid fail
 
 It is possible to set the batch size between 1 and 10.
 
+### Maximum Batching Window
+
+```yaml
+constructs:
+    my-queue:
+        # ...
+        maxBatchingWindow: 5 # SQS will wait 5 seconds (so that it can batch any messages together) before delivering to lambda
+```
+
+*Default: 60 seconds*
```

Comment:
https://github.com/getlift/lift/issues/160#issuecomment-1027352596

> Also I'm wondering if the default for maxBatchWindow should be 0 rather than 60 as it might catch people out?

Actually I agree with you… CloudFormation says the default is `0` for SQS:

<img width="881" alt="image" src="https://user-images.githubusercontent.com/720328/152156244-a47e1d68-97d5-4923-9354-29f20ebad474.png">

I would say we should switch to 0, WDYT @t-richard?

Reply:
AFAIK the batching window is mostly for cost/usage optimization. It will wait longer so it can eventually gather more messages for a batch which will avoid too much Lambda invocations.

Also, setting this to something else than `0` allows for batches of more than 10 items which we don't currently allow (but maybe we should :slightly_smiling_face:).

There's no clear answer here, it depends on the use case people make from the queue. Sometimes performance will be prefered, sometimes usage optimization. 

If I had to make a choice, I'd keep it as is. 60 seconds seems like a good default to me because it allows for better optimization without impacting performance too much. It will have no impact for batchSize of 1 and most of the time you are seeking optimization when using batches.

Reply:
Thanks, I wasn't sure if there were scenarios I was missing.

Let's use `0` as default then, because:

- it's consistent with the actual default from AWS
- it's not surprising (anything else is surprising because it makes the queue non-real-time, which isn't stated upfront in this construct)

As you mentioned, setting any other value is an optimization that users can make based on the specific usage for their queue, so let's keep it real-time by default.

Reply:
I'll update the PR with `0` as the default.